src/main/java/com/puppycrawl/tools/checkstyle/filters/XpathFilter.java

```diff
@@ -149,6 +149,7 @@ private boolean isMessageNameMatching(TreeWalkerAuditEvent event) {
      * @param event event
      * @return true is matching
      */
+    @SuppressWarnings("rawtypes")
```

Comment:
@timurt , please explain this suppression ?

where is raw type ?

Reply:
@romani 
Idea inspections were failing with `Raw use of Parameterized class` violation
This all happened because in new version of the library definition of the class `net.sf.saxon.om.Item` changed from
`public interface Item extends Sequence`
to
`public interface Item<T extends Item> extends GroundedValue<T>`.

And since we use `net.sf.saxon.sxpath.XpathExpression.evaluate()` method which returns `List<Item>`, I didn't find another way to avoid idea inspection violation other than suppressing it

Do you have any suggestions?

Reply:
Please report an issue on Saxon to update `evaluate` signature. And we need to use this as reason "suppressed till (link)" in javadoc.
As it is Idea inspection , please disable it by javadoc tag.

Reply:
Is it ok if I leave `@SuppressWarnings` annotation on test classes?

Reply:
Not ideal, we treat test code as main code , and I still believe that suppressions from static analysis tools should not polute code. Inspections do not have ability to have separate file with suppressions, so presence them in javadoc is good compromise.

Please do suppressions as separate commit of the same PR to ease revert of it in future.

Reply:
done